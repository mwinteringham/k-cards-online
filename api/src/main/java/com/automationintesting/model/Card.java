package com.automationintesting.model;

import com.automationintesting.service.CodeGenerator;
import com.fasterxml.jackson.annotation.JsonProperty;

public class Card {

    CodeGenerator codeGenerator = new CodeGenerator();

    @JsonProperty
    private String attendeeCode;
    @JsonProperty
    private String cardType;
    @JsonProperty
    private String cardCode;

    public Card() {
    }

    public Card(String attendeeCode, String cardType, String cardCode) {
        this.attendeeCode = attendeeCode;
        this.cardType = cardType;
        this.cardCode = cardCode;
    }

    public Card(String attendeeCode, String cardType) {
        this.attendeeCode = attendeeCode;
        this.cardType = cardType;
        this.cardCode = codeGenerator.createCode();
    }

    public String getCardType() {
        return cardType;
    }

    public void setCardType(String cardType) {
        this.cardType = cardType;
    }

    public String getAttendeeCode() {
        return attendeeCode;
    }

    public void setAttendeeCode(String attendeeCode) {
        this.attendeeCode = attendeeCode;
    }

    public String getCardCode() {
        return cardCode;
    }

    @Override
    public String toString() {
        return "Card{" +
                "attendeeCode='" + attendeeCode + '\'' +
                ", cardType='" + cardType + '\'' +
                ", cardCode='" + cardCode + '\'' +
                '}';
    }
}
