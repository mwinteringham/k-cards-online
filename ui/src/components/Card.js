import BoostrapCard from 'react-bootstrap/Card';
import Button from 'react-bootstrap/Button';
import { useGlobalState } from '../state/state';
import API from '../api/api';

const Card = ({type, name, code, refresh}) => {

    const [workshopCode] = useGlobalState('workshopCode');

    const deleteCard = async () => {
        const res = await API.deleteCard(workshopCode, [code]);

        if(res.status === 202){
            {refresh()}
        }
    }

    return(
        <div>
            <BoostrapCard body bg={type}>
                <BoostrapCard.Body>
                    <p>{name}</p>
                    <h3>Card details</h3>
                    <Button onClick={deleteCard}>Delete</Button>
                </BoostrapCard.Body>
            </BoostrapCard>
        </div>
    )

}

export default Card;