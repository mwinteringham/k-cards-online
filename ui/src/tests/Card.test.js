import Card from '../components/Card';
import { render, waitFor, screen, fireEvent } from '@testing-library/react';
import axios from 'axios';

let MockAdapter = require('axios-mock-adapter');

const mock = new MockAdapter(axios);

test('renders the Card component', async () => {
    const cardComponent = render(<Card type={'red'} name={'Mark'} code={'abc'} />);

    await waitFor(() => screen.getAllByText(/Mark/i)); 

    expect(cardComponent.baseElement).toMatchSnapshot();
});

test('send a delete request for Card', async () => {
    let mockTrigger = false;
    const refreshMock = () => {
        mockTrigger = true;
    }

    mock.onDelete('/workshop/empty-workshop-code/card/abc').reply(202);

    render(<Card type={'red'} name={'Mark'} code={'abc'} refresh={refreshMock}/>);

    await waitFor(() => screen.getAllByText(/Mark/i)); 
    fireEvent.click(screen.getByText(/Delete/i));

    await waitFor(() => expect(mock.history.delete.length).toBe(1));

    expect(mockTrigger).toBe(true);
});