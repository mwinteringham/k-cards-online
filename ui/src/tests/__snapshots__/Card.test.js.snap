// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Card component 1`] = `
<body>
  <div>
    <div>
      <div
        class="card bg-red"
      >
        <div
          class="card-body"
        >
          <div
            class="card-body"
          >
            <p>
              Mark
            </p>
            <h3>
              Card details
            </h3>
            <button
              class="btn btn-primary"
              type="button"
            >
              Delete
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
